src/layers/data_type_layer.cpp

```diff
@@ -1122,8 +1215,13 @@ void data_type_layer<InputTensorDataType,
                      OutputTensorDataType>::clear_prev_error_signals_()
 {
   if (!m_persistent_error_signals) {
-    for (auto& es : m_gradient_wrt_outputs)
-      es->Empty(true);
+    for (int i = 0; i < (int)m_gradient_wrt_outputs.size(); ++i) {
+      // Error signal is reused in input layer
```

Comment:
I don't understand this comment.

Reply:
After local backprop is complete, the error signals are propagated to the parent layers and then the input error signals are cleared. Since in-place layers share the memory, clearing it would clear the reused error signals as well, so we must skip this step.

Reply:
Updated comment.